test/whitened_sparse_variational.jl

```diff
@@ -0,0 +1,38 @@
+@testset "whitened_sparse_variational" begin
+    rng = MersenneTwister(123456)
+    N_cond = 5
+    N_a = 6
+    N_b = 7
+
+    # Specify prior.
+    f = GP(Matern32Kernel())
+
+    # Sample from prior.
+    x = collect(range(-1.0, 1.0; length=N_cond))
+    fx = f(x, 1e-1)
+    y = rand(rng, fx)
+
+    # Construct optimal approximate posterior.
+    fz = fx
+    q = optimal_variational_posterior(fz, fx, y)
+    Cuu = cholesky(Symmetric(cov(fz)))
+    q_ε = MvNormal(Cuu.U' \ (mean(q) - mean(fz)), Symmetric((Cuu.U' \ cov(q)) / Cuu.U))
+
+    # Check that q_ε has been properly constructed.
+    @test mean(q) ≈ mean(fz) + Cuu.U' * mean(q_ε)
```

Comment:
```suggestion
    Cuu = cholesky(Symmetric(cov(fz), :L))
    q_ε = MvNormal(Cuu.L \ (mean(q) - mean(fz)), Symmetric((Cuu.L \ cov(q)) / Cuu.L'))

    # Check that q_ε has been properly constructed.
    @test mean(q) ≈ mean(fz) + Cuu.L * mean(q_ε)
```

Reply:
Why L rather than U?

Reply:
Ah, reads more natural to me, in literature it's usually defined in terms of the lower-triangular factor L (and here you use 3x L and 1x L'=U)

Reply:
I've partially adopted this. Please resolve if you are okay with the solution.